test/controllers/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -16,86 +16,182 @@
+        bind[SubmissionService].to(submissionServiceMock),
+        bind[CustomsCacheService].to(customsCacheServiceMock)
+      )
+      .build()
+
+  private val cfg: CSRFConfig = app.injector.instanceOf[CSRFConfigProvider].get
+  private val token: String = app.injector.instanceOf[CSRFFilter].tokenProvider.generateToken
+
+  private val messagesApi: MessagesApi = app.injector.instanceOf[MessagesApi]
+  private implicit val messages: Messages = messagesApi.preferred(FakeRequest())
+
+  private trait Test {
+    implicit val headerCarrierMock = mock[HeaderCarrier]
+    reset(authConnectorMock, submissionServiceMock, customsCacheServiceMock)
+    when(customsCacheServiceMock.fetchAndGetEntry[Choice](any(), meq(Choice.choiceId))(any(), any(), any()))
+      .thenReturn(Future.successful(Some(Choice(AllowedChoiceValues.AssociateDUCR))))
+    when(customsCacheServiceMock.fetchAndGetEntry[MucrOptions](any(), meq(MucrOptions.formId))(any(), any(), any()))
+      .thenReturn(Future.successful(Some(MucrOptions("MUCR"))))
+    when(customsCacheServiceMock.fetchAndGetEntry[AssociateDucr](any(), meq(AssociateDucr.formId))(any(), any(), any()))
+      .thenReturn(Future.successful(Some(AssociateDucr("DUCR"))))
     authorizedUser()
-    withCaching(Choice.choiceId, Some(Choice(AllowedChoiceValues.AssociateDUCR)))
-    withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
-    withCaching(AssociateDucr.formId, Some(AssociateDucr("DUCR")))
   }
 
-  override def afterEach(): Unit = {
-    super.afterEach()
-    reset(mockSubmissionService, mockCustomsCacheService)
-  }
+  "Associate DUCR Summary  onGET" should {
 
-  "Associate DUCR Summary GET" should {
+    "throw incomplete application when MUCROptions cache empty" in new Test {
 
-    "throw incomplete application when MUCROptions cache empty" in {
-      withCaching(MucrOptions.formId, None)
-      assertThrows[IncompleteApplication] {
-        await(route(app, getRequest(uri)).get)
-      }
+      when(customsCacheServiceMock.fetchAndGetEntry[MucrOptions](any(), meq(MucrOptions.formId))(any(), any(), any()))
+        .thenReturn(Future.successful(None))
+
+      assertThrows[IncompleteApplication] { await(routeGet()) }
     }
 
-    "throw incomplete application when AssociateDUCR cache empty" in {
-      withCaching(AssociateDucr.formId, None)
-      assertThrows[IncompleteApplication] {
-        await(route(app, getRequest(uri)).get)
-      }
+    "throw incomplete application when AssociateDUCR cache empty" in new Test {
+
+      when(customsCacheServiceMock.fetchAndGetEntry[MucrOptions](any(), meq(AssociateDucr.formId))(any(), any(), any()))
+        .thenReturn(Future.successful(None))
+
+      assertThrows[IncompleteApplication] { await(routeGet()) }
     }
 
-    "return Ok for GET request" in {
-      val result = route(app, getRequest(uri)).get
-      status(result) must be(OK)
+    "return Ok for GET request" in new Test {
+
+      status(routeGet()) must be(OK)
```

Comment:
Previous stuff slightly easier to read. This shorter. Doesn't matter what will be the result, just want to point this out :)